brewtils/rest/system_client.py

```diff
@@ -203,17 +203,46 @@ def __init__(self, *args, **kwargs):
             )
             kwargs.setdefault("system_name", args[2])
 
-        self._system_name = kwargs.get("system_name")
-        self._version_constraint = kwargs.get("version_constraint", "latest")
-        self._default_instance = kwargs.get("default_instance", "default")
+        system_kwargs = {
+            "system_name": "name",
+            "version_constraint": "version",
+            "default_instance": "instance_name",
+            "system_namespace": "namespace",
+        }
+
+        # brewtils.plugin.CONFIG is already populated
+        # Start by assuming that the system client is targeting self:
+        target_self = True
+
+        # Now we need to figure out if any of the provided kwargs don't match
+        # some sort of loop that checks the value of system kwargs against the CONFIG
+        # If they are different, target_self = False and break
+        for key in system_kwargs.keys():
+            if kwargs.get(key) is None:
+                continue
+            if kwargs.get(key) != brewtils.plugin.CONFIG[system_kwargs[key]]:
+                target_self = False
+                break
+
+        # Now assign self._system_name, etc based on the value of target_self
+        if target_self:
+            self._system_name = brewtils.plugin.CONFIG.name or None
+            self._version_constraint = brewtils.plugin.CONFIG.version or None
+            self._default_instance = brewtils.plugin.CONFIG.instance_name or None
+            self._system_namespace = brewtils.plugin.CONFIG.namespace or ""
```

Comment:
Hmm, I don't know that we want the `or None` statements on the end here. If we're in the target_self case and the config is missing those values for some reason it'll be more helpful for the error to say that. If we set these things to `None` we're just hiding the problem, things going to break when you try to use the SystemClient.

I could maybe see this being useful for the namespace... that one is kind of weird and this might be the right thing to do there. @TheBurchLog thoughts on that?

Reply:
`CONFIG.name`, `CONFIG.version` and `CONFIG.instance_name` are only set if the local plugin is initialized. Code using `SystemClient` outside of a plugin would not have those values set. So I do not believe that we need to have the `or None` value there. Because if the value isn't set then it needs to error out. They did something wrong with their code.

Except for `CONFIG.namespace` because that doesn't need to be set.